api/orders.js

```diff
@@ -281,8 +281,223 @@ function cancelOrder(request, response, next) {
   };
 };
 
+/**
+ *  Get the most recent spapshot of the order book for a currency pair
+ *
+ *  @url
+ *  @param {RippleAddress} request.params.account - The ripple address to use as point-of-view (returns unfunded orders for this account)
+ *  @param {String ISO 4217 Currency Code + RippleAddress} request.params.base    - Base currency as currency+issuer
+ *  @param {String ISO 4217 Currency Code + RippleAddress} request.params.counter - Counter currency as currency+issuer
+ *
+ *  @query
+ *  @param {String} [request.query.limit] - Set a limit to the number of results returned
+ *
+ *  @param {Express.js Request} request
+ *  @param {Express.js Response} response
+ *  @param {Express.js Next} next
+ */
+function getOrderBook(request, response, next) {
+  var options = request.params;
+
+  Object.keys(request.query).forEach(function(param) {
+    options[param] = request.query[param];
+  });
+
+  parseOptions(options)
+  .then(validateOptions)
+  .then(getLastValidatedLedger)
+  .then(getBidsAndAsks)
+  .spread(respondWithOrderBook)
+  .catch(next);
+
+  function parseOptions(options) {
+    var baseCurrencyArray    = options.base.split('+');
+    var counterCurrencyArray = options.counter.split('+');
+
+    options.validated = true;
+    options.order_book = options.base + '/' + options.counter;
+
+    options.base = {
+      currency: (baseCurrencyArray.length >= 1 ? baseCurrencyArray[0] : ''),
+      issuer:   (baseCurrencyArray.length >= 2 ? baseCurrencyArray[1] : '')
+    };
+
+    options.counter = {
+      currency: (counterCurrencyArray.length >= 1 ? counterCurrencyArray[0] : ''),
+      issuer:   (counterCurrencyArray.length >= 2 ? counterCurrencyArray[1] : '')
+    };
```

Comment:
can you refactor these with the code in payments.js L#534-539 into a separate helper function?